src/Microsoft.IdentityModel.Abstractions/TelemetryEventDetails.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.IdentityModel.Abstractions
+{
+    /// <summary>
+    /// Details of the telemetry event.
+    /// </summary>
+    /// <remarks>
+    /// This implementation is not meant to be thread-safe. This implementation would either need to be overridden or
```

Comment:
What would happen with multiple threads?

Reply:
This is just to acknowledge that it's based on an internal Dictionary which isn't thread safe
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?redirectedfrom=MSDN&view=net-6.0#thread-safety

One could reimplement with concurrentDictionary if that was important but I don't think there would be many applications which needed concurrency for a single event (the concurrent events should usually be separated).